### PR TITLE
fix: in cluster dialer security context is non-privileged

### PR DESCRIPTION
**Pull request by matejvasek**
# Changes

- :bug: In-cluster-dialer pod security context is non-privileged now.

### CODE REVIEW

**Review by lance — COMMENTED** [review]
Nice - assuming it works with @jrangelramos tests this lgtm

**Comment by matejvasek** [comment]
@jrangelramos please try again

**Comment by matejvasek** [comment]
I find odd I have to set up it this way. Is there any factory function that would present sane defaults for me?

cc @rhuss

**Review by lkingland — APPROVED** [review]
/lgtm

File: k8s/dialer.go [hunk 1]
```diff
@@ -102,6 +102,7 @@ func (c *contextDialer) startDialerPod(ctx context.Context) (err error) {
 	if err != nil {
 		return
 	}
+	c.restConf.WarningHandler = restclient.NoWarnings{}
```
Comment (rhuss) [inline k8s/dialer.go:105]: why is this needed ?

Comment (matejvasek) [reply k8s/dialer.go:105]: that's not exactly needed, it should be rather used depending on `--verbose -v` flags
so user is not scared in non verbose mode

**Comment by matejvasek** [comment]
@rhuss is seems that my issue was related to `Sandbox` specific settings not 4.10 vs 4.11.

If I don't set the seccomp profile the there is warning on 4.11,
but if I set the seccomp profile then it is not working in `Sandbox` cluster.

File: k8s/dialer.go (hunk 1 above)
Comment (matejvasek) [reply k8s/dialer.go:105]: I just don't want to scare users.

File: k8s/dialer.go [hunk 2]
```diff
@@ -143,6 +145,12 @@ func (c *contextDialer) startDialerPod(ctx context.Context) (err error) {
 					Stdin:     true,
 					StdinOnce: true,
 					Args:      []string{"-u", "-", "OPEN:/dev/null,append"},
+					SecurityContext: &coreV1.SecurityContext{
+						Privileged:               new(bool),
+						AllowPrivilegeEscalation: new(bool),
```
Comment (rhuss) [inline k8s/dialer.go:150]: why do you need to set both to `false` ? Isn't this the default value anyway ?

File: k8s/dialer.go (hunk 1 above)
Comment (rhuss) [reply k8s/dialer.go:105]: I see, fine for me.

File: k8s/dialer.go (hunk 2 above)
Comment (matejvasek) [reply k8s/dialer.go:150]: by default it's null, IIRC it caused troubles

**Comment by matejvasek** [comment]
@rhuss 
In the end I can decide between:
* No warnings on 4.11 but totally broken `Sandbox`.
* Single warning on 4.11 but working `Sandbox`.

File: k8s/dialer.go (hunk 2 above)
Comment (matejvasek) [reply k8s/dialer.go:150]: It should be explicitly ptr to false.

**Comment by matejvasek** [comment]
By `Sandbox` I mean OpenShift evaluation instances.

File: k8s/dialer.go (hunk 2 above)
Comment (matejvasek) [reply k8s/dialer.go:150]: I would too assumed it should work with zero/null value, but it doesn't.
It also applies to `RunAsNonRoot` despite the fact the user of the image is not `root`.
